web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1432,19 +1442,41 @@
             return defer.promise;
           },
 
+          /**
+           * @ngdoc method
+           * @methodOf gn_map.addEsriRestFromScratch:gnMap
+           * @name gnMap#addEsriRestFromScratch
+           *
+           * @description
+           * Here is the method to use when you want to add a ESRIREst layer from
+           * a url and a name (layer identifier).
+           *
+           * Return a promise with ol.Layer as data is succeed, and url/name
+           * if failure.
+           * If createOnly, we don't add the layer to the map.
+           * If the md object is given, we add it to the layer, or we try
+           * to retrieve it in the catalog
+           *
+           * @param {ol.Map} map to add the layer
+           * @param {string} url of the service
+           * @param {string} name of the layer (identifier)
+           * @param {boolean} createOnly or add it to the map
+           * @param {!Object} md object
+           */
           addEsriRestFromScratch: function(map, url, name, createOnly, md) {
             if (url === '') {
               var error = "Trying to add an ESRI layer with no service URL. Layer name is " + name + ". Check the metadata or the map.";
               console.warn(error);
               return $q.reject(error);
             }
             var serviceUrl = url.replace(/(.*\/MapServer).*/, '$1');
-            var layer = !!name && parseInt(name).toString() === name
+            var layer = angular.isNumber(name)
               ? name
               : url.replace(/.*\/([^\/]*)\/MapServer\/?(.*)/, '$2');
             name = url.replace(/.*\/([^\/]*)\/MapServer\/?(.*)/, '$1 $2');
 
-            var olLayer = getTheLayerFromMap(map, name, url);
+            // Use the url and the layer identifier to check if the layer exists
+            var olLayer = getTheLayerFromMap(map, layer, url);
```

Comment:
You're switching from `layer` to `name` here, is that intentional? It looks like this could break things.

Reply:
`getTheLayerFromMap` was not supporting ESRI Rest layers, so that code was useless and should be used the `layer` that is the idenfifier, not the string value that is also unclear why is implemented  like this:

` name = url.replace(/.*\/([^\/]*)\/MapServer\/?(.*)/, '$1 $2');`

For example for https://land.discomap.eea.europa.eu/arcgis/rest/services/Agriculture/agri_extensification_on_HNVfarmland06/MapServer

sets `name` to `agri_extensification_on_HNVfarmland06 `
 
 For https://land.discomap.eea.europa.eu/arcgis/rest/services/Agriculture/agri_extensification_on_HNVfarmland06/MapServer/0
 
 sets `name` to `agri_extensification_on_HNVfarmland06 0`
 

 it's totally unclear the reason for that, but that can't be used to identify the layer in the map.